### PR TITLE
Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey'

### PR DESCRIPTION
**Pull request by marco-carvalho**
Benchmark:
```cs
[SimpleJob]
[MemoryDiagnoser]
[Orderer(SummaryOrderPolicy.FastestToSlowest)]
[RankColumn]
public class Test
{
    private Dictionary<int, string> dict = new Dictionary<int, string>();
    private int size = 1000;
    private int keyToFind = 999;

    [GlobalSetup]
    public void Setup()
    {
        for (int i = 0; i < size; i++)
        {
            dict.Add(i, "aaa");
        }
    }

    [Benchmark]
    public void ContainsKeyPlusArrayLookup()
    {
        if (dict.ContainsKey(keyToFind))
        {
            var response = dict[keyToFind];
        }
    }

    [Benchmark]
    public void TryGetValue()
    {
        dict.TryGetValue(keyToFind, out var response);
    }
}
```

Results:
```sh
|                     Method |      Mean |     Error |    StdDev |    Median | Rank | Allocated |
|--------------------------- |----------:|----------:|----------:|----------:|-----:|----------:|
|                TryGetValue |  9.649 ns | 0.3920 ns | 1.0927 ns |  9.136 ns |    1 |         - |
| ContainsKeyPlusArrayLookup | 14.196 ns | 0.1173 ns | 0.1040 ns | 14.186 ns |    2 |         - |
```

### CODE REVIEW

File: src/Ryujinx.Input/Motion/CemuHook/Client.cs [hunk 1]
```diff
@@ -338,11 +338,11 @@ public void HandleResponse(byte[] data, int clientId)
...
                             int slot = inputData.Shared.Slot;
 
-                            if (_motionData.ContainsKey(clientId))
+                            if (_motionData.TryGetValue(clientId, out var motionDataItem))
                             {
-                                if (_motionData[clientId].ContainsKey(slot))
+                                if (motionDataItem.ContainsKey(slot))
                                 {
-                                    MotionInput previousData = _motionData[clientId][slot];
+                                    MotionInput previousData = motionDataItem[slot];
```
Comment (sampletext32) [inline src/Ryujinx.Input/Motion/CemuHook/Client.cs:345]: why not also use TryGetValue on motionDataItem? Isn't it a dict as well?

Comment (marysaka) [reply src/Ryujinx.Input/Motion/CemuHook/Client.cs:345]: It seems that they are using a tool to automatically replace some kind of lints here considering their other PRs.

So I guess the tool didn't caught that.

**Review by marysaka — APPROVED** [review]
lgtm

File: src/Ryujinx.Input/Motion/CemuHook/Client.cs (hunk 1 above)
Comment (marco-carvalho) [reply src/Ryujinx.Input/Motion/CemuHook/Client.cs]: @sampletext32 could you check again and resolve it if this is fixed?

Comment (sampletext32) [reply src/Ryujinx.Input/Motion/CemuHook/Client.cs]: seems fine

**Review by AcK77 — APPROVED** [review]

**Review by riperiperi — APPROVED** [review]

**Pull request merged**